### ISSUE TITLE
Clean up deployments in dev once the PR's are closed

### ISSUE THREAD

**Issue by thegentlemanphysicist** [bcgov/digital.gov.bc.ca#295]
Currently deployments stay in openshift once the PR is closed.  I believe these should be deleted once the PR is closed.

This requires creating another stage/job

**Comment by HeatherRemacle** [comment]
Do we want to ask the community for support or advice on this?

**Comment by thegentlemanphysicist** [comment]
Current status:  The deployment created for a PR is being deleted once the PR is closed as the ticket required. 

However, the git action that runs this deletion is producing a permissions error when it does so.  I believe it is attempting to list assets it does not have the permissions for.  When I tried to fix this, it broke the permissions on other Github Actions like automatic deployment on PR's being opened.  With Patrick's help we were able to fix the deployments, but it has made me very hesitant about breaking things again if I try to fix things myself.

I'm placing this back in the backlog because I do not want to break things when the entire dev ops community is on vacation.

The error can be found in the run output in the deletion outputs:

https://github.com/bcgov/digital.gov.bc.ca/runs/1626980570?check_suite_focus=true

Below is a copy of the error message:

TASK [Delete digital-gov-frontend of PR 484 in xdwiwd-dev namespace] ***********
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "oc delete -n \"xdwiwd-dev\" all --selector pr=484\n", "delta": "0:00:07.316987", "end": "2020-12-30 16:46:08.586645", "msg": "non-zero return code", "rc": 1, "start": "2020-12-30 16:46:01.269658", "stderr": "Error from server (Forbidden): services is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list services in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): daemonsets.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list daemonsets.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): deployments.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list deployments.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): replicasets.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list replicasets.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): statefulsets.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list statefulsets.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): horizontalpodautoscalers.autoscaling is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list horizontalpodautoscalers.autoscaling in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): jobs.batch is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list jobs.batch in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): cronjobs.batch is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list cronjobs.batch in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): buildconfigs.build.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list buildconfigs.build.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): builds.build.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list builds.build.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): imagestreams.image.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list imagestreams.image.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched\nError from server (Forbidden): routes.route.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list routes.route.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched", "stderr_lines": ["Error from server (Forbidden): services is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list services in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): daemonsets.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list daemonsets.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): deployments.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list deployments.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): replicasets.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list replicasets.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): statefulsets.apps is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list statefulsets.apps in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): horizontalpodautoscalers.autoscaling is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list horizontalpodautoscalers.autoscaling in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): jobs.batch is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list jobs.batch in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): cronjobs.batch is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list cronjobs.batch in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): buildconfigs.build.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list buildconfigs.build.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): builds.build.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list builds.build.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): imagestreams.image.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list imagestreams.image.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched", "Error from server (Forbidden): routes.route.openshift.io is forbidden: User \"system:serviceaccount:xdwiwd-tools:github-cicd\" cannot list routes.route.openshift.io in the namespace \"xdwiwd-dev\": no RBAC policy matched"], "stdout": "pod \"digital-gov-frontend-pr-484-4-ktrss\" deleted\npod \"digital-gov-frontend-pr-484-4-wzj8l\" deleted\npod \"digital-gov-frontend-pr-484-4-xrdgg\" deleted\nreplicationcontroller \"digital-gov-frontend-pr-484-2\" deleted\nreplicationcontroller \"digital-gov-frontend-pr-484-3\" deleted\nreplicationcontroller \"digital-gov-frontend-pr-484-4\" deleted\ndeploymentconfig.apps.openshift.io \"digital-gov-frontend-pr-484\" deleted", "stdout_lines": ["pod \"digital-gov-frontend-pr-484-4-ktrss\" deleted", "pod \"digital-gov-frontend-pr-484-4-wzj8l\" deleted", "pod \"digital-gov-frontend-pr-484-4-xrdgg\" deleted", "replicationcontroller \"digital-gov-frontend-pr-484-2\" deleted", "replicationcontroller \"digital-gov-frontend-pr-484-3\" deleted", "replicationcontroller \"digital-gov-frontend-pr-484-4\" deleted", "deploymentconfig.apps.openshift.io \"digital-gov-frontend-pr-484\" deleted"]}
